src/pn-pubsub.ts

```diff
@@ -82,31 +130,54 @@ export class PNPubSub implements PubSubEngine {
 
   public subscribe(
     channel: string,
-    onMessage: Function,
-    options: Object
+    onMessage: OnMessage,
   ): Promise<number> {
+
     const id = this.subscribeId++;
+    this.subscriptions[id] = channel;
+
+    const refs = this.subsRefsMap[channel]
```

Comment:
Perhaps it's better to store each subscriber in another map so you don't have to iterate over an array?

Reply:
we would still have to iterate over an array because 1 channel has multiple resolvers.

so the main idea behind this pr is removing the event listener from the subscribe function to the constructor and this is important because ...
when we add an message event handler (the graphql resolver) it will be called on every message. so it was bad to register an event listener for every client subscription

this changed from (multiple event listeners) ===> a single event listeners that loops over the "correct" resolvers and call them

Reply:
Ok, thanks for explaining.   Hard to visualize w/o seeing the concrete examples but I think you will be adding tests and that will help.  

So, question,  how many clients do we have in our use case of this? Will it just be every instance of osaka (graphql gateway)?